### PR TITLE
View parameters in adaptive auth templates as code segments

### PR DESCRIPTION
**Pull request by ZiyamSanthosh**
Before:
Screenshot 2023-02-15 at 11 12 10

After:
Screenshot 2023-02-15 at 11 10 33

### Related Issues
- https://github.com/wso2-enterprise/asgardeo-product/issues/14966

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx [hunk 1]
```diff
@@ -111,6 +112,39 @@ export const TemplateDescription: FunctionComponent<TemplateDescriptionPropsInte
...
+        if (isObject(template.parametersDescription)) {
+            Object.entries(template.parametersDescription).map(([param]) => {
+                params.push(param);
+            });
+        }
+        return params;
+    }
+
+    const generatePrerequisite = (prerequisite: string, params: Array<string>): ReactElement => {
+        const sentenceArray = prerequisite.split(" ");
```
Comment (JayaShakthi97) [inline apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx:127]: Let's define the types of the variables. Applicable to other places as well.

File: apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx [hunk 2]
```diff
@@ -167,7 +202,7 @@ export const TemplateDescription: FunctionComponent<TemplateDescriptionPropsInte
                                         .map(([ param, description ], index: number) => {
                                             return (
                                                 <Table.Row key={ index }>
-                                                    <Table.Cell>{ param }</Table.Cell>
+                                                    <Table.Cell><Code>{ param }</Code></Table.Cell>
```
Comment (JayaShakthi97) [inline apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx:205]:
```suggestion
                                                    <Table.Cell>
                                                        <Code>{ param }</Code>
                                                    </Table.Cell>
```
Let's take the `<Code>` element to a newline.

File: apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx [hunk 3]
```diff
@@ -111,6 +112,39 @@ export const TemplateDescription: FunctionComponent<TemplateDescriptionPropsInte
...
     };
+
+    const getParameters = () : string[] => {
+        const params = [];
+        if (isObject(template.parametersDescription)) {
+            Object.entries(template.parametersDescription).map(([param]) => {
+                params.push(param);
+            });
+        }
+        return params;
```
Comment (JayaShakthi97) [inline apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx:123]: Let's add a newline before the return statement.
```suggestion

        return params;
```

File: apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx [hunk 4]
```diff
@@ -111,6 +112,39 @@ export const TemplateDescription: FunctionComponent<TemplateDescriptionPropsInte
...
+    }
+
+    const generatePrerequisite = (prerequisite: string, params: Array<string>): ReactElement => {
+        const sentenceArray = prerequisite.split(" ");
+        const modified = [];
+        let content = ""
+        sentenceArray.map((word, index) => {
+            if (params.includes(word)) {
+                modified.push(<span key={index}>{content} </span>)
```
Comment (JayaShakthi97) [inline apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx:132]:
```suggestion
                modified.push(<span key={ index }>{ content }</span>)
```

File: apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx [hunk 5]
```diff
@@ -111,6 +112,39 @@ export const TemplateDescription: FunctionComponent<TemplateDescriptionPropsInte
...
+                modified.push(<span key={index}><Code>{word}</Code> </span>)
```
Comment (JayaShakthi97) [inline apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx:133]:
```suggestion
                modified.push(<span key={ index }><Code>{ word }</Code></span>)
```

File: apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx [hunk 6]
```diff
@@ -111,6 +112,39 @@ export const TemplateDescription: FunctionComponent<TemplateDescriptionPropsInte
...
+            });
+        }
+        return params;
+    }
+
+    const generatePrerequisite = (prerequisite: string, params: Array<string>): ReactElement => {
+        const sentenceArray = prerequisite.split(" ");
+        const modified = [];
+        let content = ""
+        sentenceArray.map((word, index) => {
```
Comment (thivi) [inline apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx:130]: Let's define the types of word and index

File: apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx [hunk 7]
```diff
@@ -111,6 +112,39 @@ export const TemplateDescription: FunctionComponent<TemplateDescriptionPropsInte
...
+            if (params.includes(word)) {
+                modified.push(<span key={index}>{content} </span>)
+                modified.push(<span key={index}><Code>{word}</Code> </span>)
+                content = ""
+            } else {
+                content = content.concat(word + " ");
+            }
+        });
+        modified.push(<span>{content}</span>)
```
Comment (thivi) [inline apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx:139]:
```suggestion
        modified. Push(<span>{ content }</span>)
```

File: apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx [hunk 8]
```diff
@@ -111,6 +112,39 @@ export const TemplateDescription: FunctionComponent<TemplateDescriptionPropsInte
...
+        return (
+            <p>
+                <React.Fragment>
+                    {modified.map((element) => element)}
+                </React.Fragment>
```
Comment (thivi) [inline apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx:144]: Why do we need to wrap it inside a fragment?

File: apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx (hunk 3 above)
Comment (ZiyamSanthosh) [reply apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx:125]: Resolved in [0aad4b3](https://github.com/wso2/identity-apps/pull/3777/commits/0aad4b38955f51616d9d2927bdaf9c64a2dbfe54)

File: apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx [hunk 9]
```diff
@@ -111,6 +112,39 @@ export const TemplateDescription: FunctionComponent<TemplateDescriptionPropsInte
         );
 
     };
+
+    const getParameters = () : string[] => {
```
[the remark this answers is not in the document]
Comment (ZiyamSanthosh) [reply apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx]: Resolved in [0aad4b3](https://github.com/wso2/identity-apps/pull/3777/commits/0aad4b38955f51616d9d2927bdaf9c64a2dbfe54)

File: apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx (hunk 1 above)
Comment (ZiyamSanthosh) [reply apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx]: Resolved in [0aad4b3](https://github.com/wso2/identity-apps/pull/3777/commits/0aad4b38955f51616d9d2927bdaf9c64a2dbfe54)

File: apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx (hunk 2 above)
Comment (ZiyamSanthosh) [reply apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx]: Resolved in [0aad4b3](https://github.com/wso2/identity-apps/pull/3777/commits/0aad4b38955f51616d9d2927bdaf9c64a2dbfe54)

File: apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx (hunk 4 above)
Comment (ZiyamSanthosh) [reply apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx]: Resolved in [0aad4b3](https://github.com/wso2/identity-apps/pull/3777/commits/0aad4b38955f51616d9d2927bdaf9c64a2dbfe54)

File: apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx (hunk 5 above)
Comment (ZiyamSanthosh) [reply apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx]: Resolved in [0aad4b3](https://github.com/wso2/identity-apps/pull/3777/commits/0aad4b38955f51616d9d2927bdaf9c64a2dbfe54)

File: apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx (hunk 6 above)
Comment (ZiyamSanthosh) [reply apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx]: Resolved in [0aad4b3](https://github.com/wso2/identity-apps/pull/3777/commits/0aad4b38955f51616d9d2927bdaf9c64a2dbfe54)

File: apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx (hunk 7 above)
Comment (ZiyamSanthosh) [reply apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx]: Resolved in [0aad4b3](https://github.com/wso2/identity-apps/pull/3777/commits/0aad4b38955f51616d9d2927bdaf9c64a2dbfe54)

File: apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx (hunk 8 above)
Comment (ZiyamSanthosh) [reply apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx]: Resolved in [0aad4b3](https://github.com/wso2/identity-apps/pull/3777/commits/0aad4b38955f51616d9d2927bdaf9c64a2dbfe54)

File: apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx [hunk 10]
```diff
@@ -111,6 +112,45 @@ export const TemplateDescription: FunctionComponent<TemplateDescriptionPropsInte
...
+        if (isObject(template.parametersDescription)) {
+            Object.entries(template.parametersDescription).map(([ param ]) => {
+                params.push(param);
+            });
+        }
+
+        return params;
+    };
+
+    const generatePrerequisite = (prerequisite: string, params: Array<string>): ReactElement => {
```
Comment (thivi) [inline apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/template-description.tsx:128]:
```suggestion
    const generatePrerequisite = (prerequisite: string, params: string[]): ReactElement => {
```

**Review by thivi — APPROVED** [review]

**Pull request merged**